### PR TITLE
Remove dependancy on external goimports binary

### PR DESCRIPTION
**Pull request by eparis**
Instead of needed a random binary on your host to build kubernetes just do it in code.  The binary has caused problems for a number of people. Some didn't have it installed. Travis and shippable broke because it wasn't in the PATH. Someone had an old version installed on their system so they got 'bad' results.

Just make this a vendor and do it in code for consistency and predictability.

### CODE REVIEW

**Comment by eparis** [comment]
@derekwaynecarr @uluyol @mikedanese

File: cmd/genconversion/conversion.go [hunk 1]
```diff
@@ -76,13 +82,19 @@ func main() {
...
-	if err := generator.RegisterConversionFunctions(funcOut, fmt.Sprintf("%s.Scheme", apiShort)); err != nil {
+	if err := generator.RegisterConversionFunctions(data, fmt.Sprintf("%s.Scheme", apiShort)); err != nil {
 		glog.Fatalf("Error while writing conversion functions: %v", err)
 	}
+
+	b, err := imports.Process("", data.Bytes(), nil)
+	if err != nil {
+		glog.Fatalf("error while update imports: %v", err)
+	}
+	funcOut.Write(b)
```
Comment (uluyol) [inline cmd/genconversion/conversion.go]: You should check that the write is successful. Omitting it is safe for data but this one can fail.

File: cmd/gendeepcopy/deep_copy.go [hunk 2]
```diff
@@ -86,13 +98,18 @@ func main() {
...
 	}
-	if err := generator.RegisterDeepCopyFunctions(funcOut, registerTo); err != nil {
+	if err := generator.RegisterDeepCopyFunctions(data, registerTo); err != nil {
 		glog.Fatalf("error while registering deep copy functions: %v", err)
 	}
+	b, err := imports.Process("", data.Bytes(), nil)
+	if err != nil {
+		glog.Fatalf("error while update imports: %v", err)
+	}
+	funcOut.Write(b)
```
Comment (uluyol) [inline cmd/gendeepcopy/deep_copy.go]: Likewise, check for an error.

File: hack/after-build/update-generated-conversions.sh [hunk 3]
```diff
@@ -33,26 +33,15 @@ function generate_version() {
 
 	sed 's/YEAR/2015/' hack/boilerplate/boilerplate.go.txt > "$TMPFILE"
 	cat >> "$TMPFILE" <<EOF
-package ${version##*/}
+// THIS ENTIRE FILE IS AUTO-GENERATED BY \$KUBEROOT/hack/update-generated-conversions
```
Comment (uluyol) [inline hack/after-build/update-generated-conversions.sh]: Nit: Add a DO NOT EDIT note?

Comment (eparis) [inline hack/after-build/update-generated-conversions.sh]: Will do.

**Comment by eparis** [comment]
updated to address comments.

File: cmd/genconversion/conversion.go [hunk 4]
```diff
@@ -58,9 +60,16 @@ func main() {
 		funcOut = file
 	}
 
+	data := new(bytes.Buffer)
+
 	group, version := path.Split(*groupVersion)
 	group = strings.TrimRight(group, "/")
 
+	_, err := data.WriteString(fmt.Sprintf("package %v\n", version))
```
Comment (uluyol) [inline cmd/genconversion/conversion.go]: We are guaranteed that err will be nil, but I don't mind if the checks stays.

**Comment by uluyol** [comment]
LGTM

**Comment by eparis** [comment]
adding LGTM flag for @uluyol

**Pull request merged**